plugins/in_zmq/in_zmq.c

```diff
@@ -0,0 +1,296 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  ZMQ input plugin for Fluent Bit
+ *  ===============================
+ *  Copyright (C) 2019      The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_input.h>
+#include <fluent-bit/flb_utils.h>
+#include <fluent-bit/flb_engine.h>
+#include <fluent-bit/flb_pack.h>
+#include <fluent-bit/flb_error.h>
+#include <msgpack.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <czmq.h>
+
+#include "in_zmq.h"
+
+#ifdef __GNUC__
+#define likely(x)       __builtin_expect(!!(x), 1)
+#define unlikely(x)     __builtin_expect(!!(x), 0)
+#else
+#define likely(x)       (x)
+#define unlikely(x)     (x)
+#endif
+
+static int zmq_config_read(struct flb_in_zmq_ctx *ctx,
+                           struct flb_input_instance *i_ins)
+{
+    const char *hwm_str;
+
+    /* Get input properties */
+    ctx->zmq_endpoint = flb_input_get_property("endpoint", i_ins);
```

Comment:
Probably you should use standard parameter names (`Listen` and
`port` for TCP and `Path` for UNIX sock) instead of `endpoint`.

See: https://docs.fluentbit.io/manual/pipeline/inputs/syslog